src/librustc_metadata/diagnostics.rs

```diff
@@ -26,6 +26,18 @@ name. Example:
 ```
 "##,
 
+E0455: r##"
+A framework link kind was used and the OS used to compile isn't the OSX.
+Erroneous code example:
+
+```compile_fail"
+#[link(name = "FooCoreServices",  kind = "framework")] extern {}
+// OS used to compile is Linux for example
+```
+
+When using 'framework' link kind, please ensure to use OSX to compile the code.
```

Comment:
Actually, I would leave this part out and just talk about conditional compilation, because it's the right fix.